### PR TITLE
[MM-56295] Convert `./components/post_view/message_attachments/message_attachment_list.tsx` from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Summary

Convert `./components/post_view/message_attachments/message_attachment_list.tsx` from Class Component to Function Component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25757
Jira https://mattermost.atlassian.net/browse/MM-56295

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/post_view/message_attachments/message_attachment_list.tsx [hunk 1]
```diff
@@ -33,32 +33,26 @@ type Props = {
...
-                id={`messageAttachmentList_${this.props.postId}`}
-                className='attachment__list'
-            >
-                {content}
-            </div>
-        );
-    }
-}
+const MessageAttachmentList = ({
+    imagesMetadata = {},
```
Comment (larkox) [inline webapp/channels/src/components/post_view/message_attachments/message_attachment_list.tsx:37]: To make sure the default is always the same, you can define outside of the component something like `const EMPTY_METADATA: Record<string, PostImage> = {};` and use that here. That way, the default is always the same reference, and you avoid potential re-renders down the line.

**Review by larkox — CHANGES_REQUESTED** [review]
One minor change, but looks great!

**Review by larkox — APPROVED** [review]

**Review by harshilsharma63 — APPROVED** [review]
👌👌👌

**Pull request merged**